### PR TITLE
migrate to IMDSv2

### PR DESCRIPTION
**Pull request by mtanda**
Use new Instance Metadata Service to get metadata.

https://aws.amazon.com/blogs/security/defense-in-depth-open-firewalls-reverse-proxies-ssrf-vulnerabilities-ec2-instance-metadata-service/

This change will help to disable IMDSv1 in the public instance.
To keep compatible with ec2-classic environment, I add `imdsv2` option.

### CODE REVIEW

**Comment by mtanda** [comment]
The GitLab check seems to not work.

**Comment by cosmo0920** [comment]
Any chance to add testcases using VCR cassettes?

**Comment by mtanda** [comment]
@cosmo0920 I added test. Would you review the changes?

File: test/plugin/test_out_ec2_metadata.rb [hunk 1]
```diff
@@ -62,6 +62,40 @@ def create_driver(conf=CONFIG)
     end
   end
 
+  test 'configure-vpc-with-imdsv2' do
+    VCR.use_cassette('ec2-vpc-with-imdsv2', :allow_playback_repeats => true) do
+      c = %[
+        output_tag ${instance_id}.${tag}
+        aws_key_id aws_key
+        aws_sec_key aws_sec
```
Comment (cosmo0920) [inline test/plugin/test_out_ec2_metadata.rb]:
```aconf
imdsv2 true
```

is needed?

Comment (mtanda) [reply test/plugin/test_out_ec2_metadata.rb]: Oops, yes. I forgot to change it.

**Comment by mtanda** [comment]
@cosmo0920 Thanks for review :-)
Would you check this again?

**Comment by cosmo0920** [comment]
@takus GitLab CI seems not to be working. Could you take a look?

**Comment by takus** [comment]
@mtanda @cosmo0920 
Thanks for letting me know. I temporally disabled the check by Gitlab. Since this pull request doesn't touch any dependencies, the risk for disabling the check is low. I'll investigate the Gitlab issue soon and enable it again.

**Comment by cosmo0920** [comment]
OK. Then, let's merge this PR!
Thank you for your contribution, @mtanda !

**Comment by mtanda** [comment]
Thank you for merge and help!

**Pull request merged**